src/Engine/Manager/RenderManager.cpp

```diff
@@ -226,6 +226,22 @@ void RenderManager::Render(World& world, const glm::mat4& translationMatrix, con
 
     const std::vector<Mesh*>& meshComponents = meshes.GetAll();
 
+    {
+        PROFILE("Render z-pass meshes");
+        for (Mesh* mesh : meshComponents) {
+            if (mesh->IsKilled() || !mesh->entity->enabled)
+                continue;
+
+            if (mesh->geometry != nullptr && mesh->geometry->GetType() == Video::Geometry::Geometry3D::STATIC) {
+                Entity* entity = mesh->entity;
+                Material* material = entity->GetComponent<Material>();
```

Comment:
Material is not relevant for a Z pre-pass, so we should skip it entirely unless there is some implicit assumption that depends on its existance, in which case that should be clearly commented.